File: Quicksort.js
//Author: Harry Wang
//Date: May 26 2021
//Description: Same as my python version, but in javascript. Requires Node.js.

const readline = require('readline');

function quick_sort(data) {
    //console.log("data");
    //console.log(data);
    var l, pivot, item
    var greater_list = [];
    var lower_list = [];
    l = data.length;
    if(l <= 1) {
        return data;
    } else {
        pivot = data.pop();
    }
    for(var i = 0; i < l-1; i++) {
        item = data[i];
        if(item > pivot) {
            //console.log(i.toString()+" Greater: "+item.toString());
            greater_list.push(item);
        } else {
            //console.log(i.toString()+" Lower: "+item.toString());
            lower_list.push(item);
        }
    }
    var pivot_list = [pivot];
    return quick_sort(lower_list).concat(pivot_list).concat(quick_sort(greater_list));
}
const aIn = readline.createInterface({
    input: process.stdin,
    output: process.stdout
  });

aIn.question('Input list with integers, all elements seperated by a space\n', arrayInput => {
    console.log(quick_sort(arrayInput.split(" ").map(x => parseInt(x))));
    aIn.close();
});

File: Quicksort.py
#Author: Harry Wang
#Date: May 9, 2021
#Description: Quick sort function in python. Inputs a list of integers and sorts them lowest to greatest. Learned from: https://www.youtube.com/watch?v=kFeXwkgnQ9U
def quick_sort(data):
    l = len(data)
    if l <= 1:
        return data
    else:
        pivot = data.pop()
    greater_list = []
    lower_list = []
    for item in data:
        if item > pivot:
            greater_list.append(item)
        else:
            lower_list.append(item)
    return quick_sort(lower_list) + [pivot] + quick_sort(greater_list)
print("Input list with integers, all elements seperated by a space")
print(quick_sort(list(map(int,input().split()))))